Convert to System.Text.Json

joeloff opened the pull request:
This PR removes the Newtonsoft.Json dependency. Additionally, the library now only support .NET Standard 2.0

joeloff commented:
@jamshedd and @ashnaga as an FYI.

joeloff commented on eng/Versions.props:
```diff
@@ -70,7 +70,7 @@
     <MicrosoftNETCoreILAsmVersion>6.0.0-rc.2.21451.6</MicrosoftNETCoreILAsmVersion>
     <MicrosoftNETSdkILVersion>6.0.0-rc.2.21451.6</MicrosoftNETSdkILVersion>
     <!-- Libraries dependencies -->
-    <SystemTextJsonVersion>6.0.0-rc.2.21451.6</SystemTextJsonVersion>
```
@NikolaMilosavljevic nothing else in the repo seems to be referencing this package, so I bumped the version to be the latest 6.0 release.

NikolaMilosavljevic replied:
This is OK - thanks for updating this package version.

MichaelSimons commented on eng/Versions.props:
```diff
@@ -70,7 +70,7 @@
     <MicrosoftNETSdkILVersion>6.0.0-rc.2.21451.6</MicrosoftNETSdkILVersion>
     <!-- Libraries dependencies -->
-    <SystemTextJsonVersion>6.0.0-rc.2.21451.6</SystemTextJsonVersion>
+    <SystemTextJsonVersion>6.0.6</SystemTextJsonVersion>
```
Any reason no to take a dependency on a 7.0 version?  From a source-build perspective this would be better.

joeloff replied:
I was just looking for a stable version, but I don't have any concerns trying out 7.0-rc1 for now and possibly the 7.0 RTM copy once we get ready to stabilize the releases library.

joeloff replied:
I updated to 7.0.0-rc.1 for now and we can update again once we have 7.0.0 RTM

joeloff commented:
Seems like moving to 7.0 breaks all the tests. Doesn't seem like we have PR builds for the repo - I'm just running locally and on the internal build.

joeloff commented:
Found the problem The JSON contains invalid data, e.g. strings with versions that have leading space and added a converter that can trim this so we have more tolerance. I also added a test to download/parse the latest JSON files that are live.

joeloff commented:
/azp run
